File: src/com/colacelli/irclib/connection/listeners/OnChannelModeListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.actors.Channel;
import com.colacelli.irclib.connection.Connection;

public interface OnChannelModeListener extends Listener {
    Type TYPE = Type.CHANNEL_MODE;

    void onChannelMode(Connection connection, Channel channel, String mode);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnChannelModeListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.actors.Channel
import com.colacelli.irclib.connection.Connection

interface OnChannelModeListener : Listener {
    fun onChannelMode(connection: Connection, channel: Channel, mode: String, vararg args: String)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.CHANNEL_MODE
    }
}
